crates/admin/src/cluster_controller/service.rs

```diff
@@ -525,6 +545,14 @@ impl<T: TransportConnect> Service<T> {
                 self.create_partition_snapshot(partition_id, response_tx)
                     .await;
             }
+            ClusterControllerCommand::ForceSeal {
+                log_id,
+                response_tx,
+            } => {
+                info!(%log_id, "Forcing seal of loglet");
+                let result = state.force_seal_log(log_id);
+                let _ = response_tx.send(result);
```

Comment:
Instead of going through the state and eventually the logs controller, could we run a `BifrostAdmin::seal_and_extend` command with the current logs configuration? I guess one would need to call `try_provisioning` from the `logs_controller` module. That way, we wouldn't need to run the `LogsController` if we are a follower.

Reply:
The reason I did it this way (and might be unnecessary) is because sealing can take long time until it finishes (because some nodes might be down, etc...) and hence I didn't want to run it in the foreground. Also to avoid handling sealing failure, etc.. and everything that LogsController takes care of already. By the time the sealing is finished the cluster state might also has changed hence IMHO it's better to decide on the next segment params after the seal not before.

I also noticed that seal_and_extend_chain tends to panic because of this line https://github.com/restatedev/restate/blob/main/crates/bifrost/src/bifrost_admin.rs#L106 which I told @AhmedSoliman about already.

Reply:
I guess we could run this this operation also as a task in the background to avoid the blocking.